metaflow/plugins/kfp/kfp.py

```diff
@@ -307,14 +308,15 @@ def _create_workflow_yaml(
         else:
             raise NotImplementedError(f"Unsupported output format {output_format}.")
 
-        workflow["spec"]["synchronization"] = {
-            "semaphore": {
-                "configMapKeyRef": {
-                    "name": sanitize_k8s_name(self.name),
-                    "key": "max_run_concurrency",
+        if not KFP_ARGO_COMPILE_WORKFLOW_ONLY:
```

Comment:
Looks like I missed earlier PRs that introduced this feature, just to make sure I'm understanding how we're using this feature, looks like we're using [Workflow-level synchronization](https://argoproj.github.io/argo-workflows/synchronization/) right? So what this means in practice is if this `Workflow` runs in any capacity (either ad-hoc, triggered by an event or from a `CronWorkflow`) it'd be bound to the `max_run_concurrency`?

Reply:
Yes! and the synchronization semaphore is shared between (WorkflowTemplate triggers,  CronWorkflow runs, and ad-hoc Workflow submissions)